### PR TITLE
Add github action to manage self-hosted runner

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

**Review by oap75 — APPROVED** [review]

**Comment by oap75** [comment]
We could use terraform too. (it's speed is much more better than doctl)

File: .github/workflows/create-runner.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        if: failure()
+        run: |
+          doctl compute droplet delete build-runner --force
+
+      - name: Remove artifacts
+        run: |
+          rm build-runner build-runner.pub init.sh || true
+
+  delete-runner:
+    if: github.event.inputs.action == 'Delete' || || github.event.workflow_run.status == 'completed'
```
Comment (F3Joule) [inline .github/workflows/create-runner.yml:113]:
```suggestion
    if: github.event.inputs.action == 'Delete' || github.event.workflow_run.status == 'completed'
```

File: .github/workflows/create-runner.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        with:
+          github-token: ${{secrets.GH_PAT}}
+          result-encoding: string
+          script: |
+            const runner = await github.actions.listSelfHostedRunnersForRepo({
+              repo: context.repo.repo,
+              owner: context.repo.owner
+            })
+            const runner_id = runner.data.runners[0].id
+            console.log(`Deleting runner ${runner_id}`)
```
Comment (F3Joule) [inline .github/workflows/create-runner.yml:129]: Probably redundant, or either can be debug log.

```suggestion
```

**Review by oap75 — CHANGES_REQUESTED** [review]
We could have our scripts in the repo and just call them in action .

**Review by siman — APPROVED** [review]

**Review by olehmell — APPROVED** [review]

**Pull request merged**